libs/ci/github/github_utils.go

```diff
@@ -0,0 +1,39 @@
+package github
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/google/go-github/v61/github"
+)
+
+func ListGithubRepos(client *github.Client) ([]*github.Repository, error) {
+	allRepos := make([]*github.Repository, 0)
+	//err := c.BindJSON(&request)
+	opts := &github.RepositoryListByOrgOptions{
+		ListOptions: github.ListOptions{PerPage: 100},
+	}
+
+	countLimit := 0
+	for {
+		opt := &github.ListOptions{Page: opts.Page, PerPage: 100}
+		listRepos, resp, err := client.Apps.ListRepos(context.Background(), opt)
+		if err != nil {
+			slog.Error("Failed to list existing repositories",
+				"error", err,
+			)
+			return nil, err
+		}
+		allRepos = append(allRepos, listRepos.Repositories...)
+		countLimit++
+		if countLimit == 2000 {
```

Comment:
2000 is too high, we are returning 100 per page . by the time we reach 2000 count it will be 200,000 repos. I'd set the limit to 20 max (so its like 2000 repos)